In [1]:
# Bibliotecas necessárias
import pandas as pd
import numpy as np
import statistics as st

import warnings
warnings.filterwarnings("ignore") # not to display warnings

In [5]:
# Load the data
df=pd.read_csv(r'C:\Users\User\Downloads\winequality-red.csv', sep=';') # sep=';' -> definir separador correto dos dados (;)
# Explore the file
df.info()
df.head() # -> primeiras 5 linhas
# df.tail() -> últimas 5 linhas

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1599 non-null   float64
 1   volatile acidity      1599 non-null   float64
 2   citric acid           1599 non-null   float64
 3   residual sugar        1599 non-null   float64
 4   chlorides             1599 non-null   float64
 5   free sulfur dioxide   1599 non-null   float64
 6   total sulfur dioxide  1599 non-null   float64
 7   density               1599 non-null   float64
 8   pH                    1599 non-null   float64
 9   sulphates             1599 non-null   float64
 10  alcohol               1599 non-null   float64
 11  quality               1599 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 150.0 KB


,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality
0,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5
1,7.8,0.88,0.00,2.6,0.098,25.0,67.0,0.9968,3.20,0.68,9.8,5
2,7.8,0.76,0.04,2.3,0.092,15.0,54.0,0.9970,3.26,0.65,9.8,5
3,11.2,0.28,0.56,1.9,0.075,17.0,60.0,0.9980,3.16,0.58,9.8,6
4,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5


O dataset tem 1599 linhas e 12 colunas.

As variáveis são fixed acidity, volatile acidity, citric acid, residual sugar, chlorides, free sulfur dioxide, total sulfur dioxide, density, pH, sulphates, alcohol e quality.

Existe 1 tipo de variável:

- Variáveis quantitativas (numéricas): fixed acidity, volatile acidity, citric acid, residual sugar, chlorides, free sulfur dioxide, total sulfur dioxide, density, pH, sulphates, alcohol e quality

Não existem valores em falta.

In [13]:
# Obtenha um resumo das estatísticas descritivas básicas de todas as variáveis numéricas em winequality-red.csv.
df.describe() 

,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality
count,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000,1599.000000
mean,8.319637,0.527821,0.270976,2.538806,0.087467,15.874922,46.467792,0.996747,3.311113,0.658149,10.422983,5.636023
std,1.741096,0.179060,0.194801,1.409928,0.047065,10.460157,32.895324,0.001887,0.154386,0.169507,1.065668,0.807569
min,4.600000,0.120000,0.000000,0.900000,0.012000,1.000000,6.000000,0.990070,2.740000,0.330000,8.400000,3.000000
25%,7.100000,0.390000,0.090000,1.900000,0.070000,7.000000,22.000000,0.995600,3.210000,0.550000,9.500000,5.000000
50%,7.900000,0.520000,0.260000,2.200000,0.079000,14.000000,38.000000,0.996750,3.310000,0.620000,10.200000,6.000000
75%,9.200000,0.640000,0.420000,2.600000,0.090000,21.000000,62.000000,0.997835,3.400000,0.730000,11.100000,6.000000
max,15.900000,1.580000,1.000000,15.500000,0.611000,72.000000,289.000000,1.003690,4.010000,2.000000,14.900000,8.000000


Variável quality:
- Média: 5.6
- Mediana: 6.0
- Desvio padrão: 0.8

Variável alcohol:
- Mínimo: 8.40
- Máximo: 14.90

Variável pH:
- Média: 3.311

Na variável quality, a média é muito próxima da mediana, o que mostra que a maioria dos vinhos apresenta qualidade média, existindo poucos vinhos com qualidade muito baixa ou muito elevada.

In [22]:
df.quality.value_counts()

quality
5    681
6    638
7    199
4     53
8     18
3     10
Name: count, dtype: int64

A qualidade mais frequente é 5, o que indica que a maioria dos vinhos apresenta qualidade média. Existem poucos vinhos com qualidade muito baixa ou muito elevada.

In [25]:
# Teor alcoólico médio por nível de qualidade
df.groupby('quality')['alcohol'].mean()

quality
3     9.955000
4    10.265094
5     9.899706
6    10.629519
7    11.465913
8    12.094444
Name: alcohol, dtype: float64

De um modo geral, observa-se que, quanto maior a qualidade do vinho, maior tende a ser o teor alcoólico.

No entanto, esta relação não é perfeitamente linear, pois existem algumas variações (por exemplo, entre as qualidades 4 e 5).

In [29]:
# Acidez volátil média por nível de qualidade
df.groupby('quality')['volatile acidity'].mean()

quality
3    0.884500
4    0.693962
5    0.577041
6    0.497484
7    0.403920
8    0.423333
Name: volatile acidity, dtype: float64

Os vinhos com menor acidez volátil tendem a ter melhor qualidade.

In [45]:
# Selecionar vinhos com qualidade maior ou igual a 7
df[df['quality']>= 7]

,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality
7,7.3,0.65,0.00,1.2,0.065,15.0,21.0,0.99460,3.39,0.47,10.00,7
8,7.8,0.58,0.02,2.0,0.073,9.0,18.0,0.99680,3.36,0.57,9.50,7
16,8.5,0.28,0.56,1.8,0.092,35.0,103.0,0.99690,3.30,0.75,10.50,7
37,8.1,0.38,0.28,2.1,0.066,13.0,30.0,0.99680,3.23,0.73,9.70,7
62,7.5,0.52,0.16,1.9,0.085,12.0,35.0,0.99680,3.38,0.62,9.50,7
...,...,...,...,...,...,...,...,...,...,...,...,...
1541,7.4,0.25,0.29,2.2,0.054,19.0,49.0,0.99666,3.40,0.76,10.90,7
1544,8.4,0.37,0.43,2.3,0.063,12.0,19.0,0.99550,3.17,0.81,11.20,7
1549,7.4,0.36,0.30,1.8,0.074,17.0,24.0,0.99419,3.24,0.70,11.40,8
1555,7.0,0.56,0.17,1.7,0.065,15.0,24.0,0.99514,3.44,0.68,10.55,7


In [47]:
# Nº de vinhos com qualidade maior ou igual a 7
df[df['quality']>= 7].shape

(217, 12)

O número de vinhos com qualidade maior ou igual a 7 é 217.

In [50]:
# Álcool Médio quando qualidade é maior ou igual a 7
round(df[df['quality']>= 7]['alcohol'].mean(), 2)

11.52

O álcool médio nos vinhos com qualidade maior ou igual a 7 é 11.52, sendo superior à média geral (aproximadamente 10.42).

In [59]:
# Vinhos com teor alcoólico superior a 14
df[df['alcohol']>14]

,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality
652,15.9,0.36,0.65,7.5,0.096,22.0,71.0,0.9976,2.98,0.84,14.9,5


In [61]:
# Nº de vinhos com teor alcoólico superior a 14
df[df['alcohol']>14].shape

(1, 12)

O número de vinhos com teor alcoólico superior a 14 é 1.

Existem muito poucos vinhos com teor alcoólico muito elevado (superior a 14), sendo apenas um caso, o que indica que valores extremos de álcool são raros.